docs/NETWORK-ACLS.md

```diff
@@ -0,0 +1,288 @@
+  with value `domain.com` will also apply to the endpoint `subdomain.domain.com`.
+  Can be combined with other match types except for `eidset` (see [limitations](#limitations)).
+
+* `eidset`: special match type for the overlay network. Matches IPs of all applications deployed in the same network
+  as well as all IPs with statically configured DNS entries (under the config field `NetworkInstanceConfig.Dns`).
+  For this type, `value` field is not used.
+  Can be combined with other match types except for `host` (see [limitations](#limitations)).
+
+* `protocol`: `value` should specify the protocol to match.
+  Protocol can be one of `tcp`, `udp`, `icmp`, or `all`, or it can be a numeric value, representing one of these
+  protocols or a different one. A protocol name from `/etc/protocols` is also allowed.
+  Protocol match can be combined with any other match type (often combined with port numbers).
+
+* `lport`: `value` should be an application *local* port number. For filtering actions, this is the source port for outbound
+  traffic and destination port for inbound traffic. For `PORTMAP` action, this represents application port as exposed
+  to the external network (i.e., if `<edge-node-ip>:2222` is mapped to `<app-ip>:22`, `lport` refers to `2222`).
+  `lport` can be combined with any other match type. It is actually required to combine `lport` and `protocol` inside
+  the same ACE. In other words, port without protocol is not valid.
+
+* `fport`: `value` should be a remote endpoint port number (*foreign* port). Used for filtering actions,
+  but not for `PORTMAP` (do not confuse with `lport`, which is still used to represent the forwarded port - the forwarded
+  port is still considered as local).
+  `fport` can be combined with any other match type. It is actually required to combine `fport` with `protocol` inside
+  the same ACE. In other words, port without protocol is not valid.
+
+## Limitations
+
+Here is a summary of all limitations of the current ACL implementation:
+
+* even though the configuration model allows to assign multiple actions to a single ACE, the current implementation
+  expects only one action and rejects any combination of actions with an error message
+
+* match types `eidset` and `host` are not supported for switch network. This is because these match types work
+  in combination with the local DNS service, which is not used for switch networks.
+
+* `host` match type does not work for domains with statically configured DNS entries. Consider using `eidset` instead,
+  which, however, matches *all* statically configured domains and also IPs of all apps in the same network.
+
+* flows dropped by ACLs within switch networks are not flow-logged. This is because denied flows which are not routed,
+  and only forwarded, get dropped before a conntrack entry is created, which is needed for a flowlog entry
```

Comment:
Conntrack entry is created in this case, but the packets do not enter routing code to be punted out of flow-mon-dummy interface. So we drop those packets explicitly. And the conntrack entry gets deleted with the drop action.

Reply:
Actually, if for example application using a switch network is only allowed to access say 150.150.0.0/16, the DROP rule will be installed in the PREROUTING of the **raw** table:
```
(ns: pillar) 2aaab4cb-e30b-485f-abb4-6e92c2aaf1e5:/# iptables -L -v -t raw
Chain PREROUTING (policy ACCEPT 3078 packets, 336K bytes)
 pkts bytes target     prot opt in     out     source               destination         
    2   656 ACCEPT     udp  --  eth1   any     anywhere             anywhere             PHYSDEV match --physdev-in nbu2x1+ match-set ipv4.local dst udp dpt:bootps
    0     0 ACCEPT     udp  --  eth1   any     anywhere             anywhere             PHYSDEV match --physdev-in nbu2x1+ udp dpt:bootps
    0     0 ACCEPT     udp  --  eth1   any     anywhere             anywhere             PHYSDEV match --physdev-in nbu2x1+ udp dpt:domain
    0     0 ACCEPT     tcp  --  eth1   any     anywhere             anywhere             PHYSDEV match --physdev-in nbu2x1+ tcp dpt:domain
    0     0 ACCEPT     all  --  eth1   any     anywhere             150.150.0.0/16       PHYSDEV match --physdev-in nbu2x1+
   72  6048 LOG        all  --  eth1   any     anywhere             anywhere             PHYSDEV match --physdev-in nbu2x1+ LOG level error prefix "FORWARD:FROM:"
   72  6048 DROP       all  --  eth1   any     anywhere             anywhere             PHYSDEV match --physdev-in nbu2x1+
```
And because raw is processed before connection tracking, I do not think that conntrack entry is created in this case.